src/ape_ethereum/provider.py

```diff
@@ -917,6 +917,11 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
             # a VM error.
             receipt.raise_for_status()
 
+        # TODO: Optional configuration?
+        if txn.data:
```

Comment:
Need something better since not all contract executions will have data (fallback, for instance).  I guess this will require an RPC call for `eth_getCode()` to verify that it's a contract.  I bet there's some useful utils somewhere for this, maybe as part of the `Address` type?

Reply:
yep, there is a `is_contract` property available on `ContractInstance` via the `Address` base.

Reply:
Abstraction spelunking... 

- [`ape.types.address.AddressType`](https://github.com/ApeWorX/ape/blob/main/src/ape/types/address.py)
- [`eth_pydantic_types.address.Address`](https://github.com/ApeWorX/eth-pydantic-types/blob/main/eth_pydantic_types/address.py)
- [`eth_typing.ChecksumAddress`](https://github.com/ethereum/eth-typing/blob/main/eth_typing/evm.py)
- [`ape.api.address.Address`](https://github.com/ApeWorX/ape/blob/2ef1d7901126120d171b5681f5ea57f3c0350b67/src/ape/api/address.py#L183) (and `BaseAddress`)

:sweat_smile: